main/main.cpp

```diff
@@ -1141,13 +1141,16 @@ Error Main::setup2(Thread::ID p_main_tid_override) {
 	GLOBAL_DEF("application/config/icon", String());
 	ProjectSettings::get_singleton()->set_custom_property_info("application/config/icon", PropertyInfo(Variant::STRING, "application/config/icon", PROPERTY_HINT_FILE, "*.png,*.webp"));
 
-	if (bool(GLOBAL_DEF("display/window/handheld/emulate_touchscreen", false))) {
-		if (!OS::get_singleton()->has_touchscreen_ui_hint() && Input::get_singleton() && !(editor || project_manager)) {
-			//only if no touchscreen ui hint, set emulation
-			InputDefault *id = Object::cast_to<InputDefault>(Input::get_singleton());
-			if (id)
+	InputDefault *id = Object::cast_to<InputDefault>(Input::get_singleton());
+	if (id) {
+		if (bool(GLOBAL_DEF("display/window/handheld/emulate_touchscreen", false)) && !(editor || project_manager)) {
+			if (!OS::get_singleton()->has_touchscreen_ui_hint()) {
+				//only if no touchscreen ui hint, set emulation
 				id->set_emulate_touch(true);
+			}
 		}
+
+		id->set_emulate_mouse_from_touch(bool(GLOBAL_DEF("display/window/handheld/emulate_mouse_from_touch", true)));
```

Comment:
Remember that we now have a tag system, so this setting could be tagged for mobile with different values, for example:

GLOBAL_DEF("display/window/handheld/emulate_mouse_from_touch", false)
GLOBAL_DEF("display/window/handheld/emulate_mouse_from_touch.mobile", true)

will only make it work on platforms that are mobile

Reply:
For 3.0 I've submitted a dedicated PR (#18082) with the defaults suggested by @reduz, to match how stuff worked there.

For 3.1, I think it's better to take the step of letting _everything work with everything_ by default, not considering dektop touch any more special than mobile touch.

Reply:
Well, mouse-to-touch may be better off by default because it may produce surprising effects if user is not aware of it.

Reply:
Did you mean touch-to-mouse? Because for me it was a very surprising effect when I *couldn't* interact with Godot GUIs using my touch screen even though I could with my mouse. On desktop platforms, I would have expected a touch screen to act the same as a mouse (so I could use a touch screen to interact with Godot GUI widgets).

Reply:
What I'm proposing disabled by default is _emulate touch from mouse_. _Emulate mouse from touch_ is what you -and probably most people- expect to be working by default, and it will be.

(The setting is in the form _emulate_<result>_from_<source>_. I gave this some thought and I believe, although it looks as it's harder to understand, it avoids the alternatives, which would be _convert/translate_<source>_to_<result>_. The concept of conversion of events is a bit lower level than emulation, which is a concept everybody is familiar with. In my comment I've written it in the direct form, so maybe that's been the source of confusion.)